test/e2e/pods.go

```diff
@@ -289,29 +291,38 @@ var _ = Describe("Pods", func() {
 		pods, err := podClient.List(labels.SelectorFromSet(labels.Set(map[string]string{"time": value})), fields.Everything())
 		Expect(len(pods.Items)).To(Equal(1))
 
-		By("retrieving the pod")
-		podOut, err := podClient.Get(pod.Name)
-		if err != nil {
-			Fail(fmt.Sprintf("Failed to get pod: %v", err))
-		}
+		// Standard get, update retry loop
+		expectNoError(wait.Poll(time.Second*5, time.Minute*2, func() (bool, error) {
+			By("retrieving the pod")
+			podOut, err := podClient.Get((*pod).Name)
+			if err != nil {
+				return false, fmt.Errorf("Failed to get pod: %v", err)
+			}
 
-		By("updating the pod")
-		value = "time" + value
-		pod.Labels["time"] = value
-		pod.ResourceVersion = podOut.ResourceVersion
-		pod.UID = podOut.UID
-		pod.Spec.Host = podOut.Spec.Host
-		pod, err = podClient.Update(pod)
-		if err != nil {
-			Fail(fmt.Sprintf("Failed to update pod: %v", err))
-		}
+			By("updating the pod")
+			value = "time" + value
+			pod.Labels["time"] = value
```

Comment:
Agreed.  That's not what the original test did, so I tried to minimize changes.  But in retrospect I agree and I'll fix that.